src/controller/binding.hxx

```diff
@@ -52,6 +57,10 @@ public:
 	/// maps from Gridlogic::State to MIDI output value from binding
 	std::map<int,int> clipStateMap;
 
+	/// arbitrary-length data (TODO use this instead of data)
+	unsigned char *dataList;
+	unsigned int dataListSize;
+
```

Comment:
I think this is still a little ugly, prehaps this could be merged with `data`. Then all instances of `data` could receive either a single value or an array.

Reply:
Shall I try to combine `data`, `dataList` and `dataListSize`?

When reading `data` has a single value, it is assumed to be of size one (i.e. `"data": 5` is assumed to be equivalent to `"data": [5]`). When writing, a single element is writting as `"data": 5` whereas multiple elements are written as an array.